bindata/network/ovn-kubernetes/managed/alert-rules.yaml

```diff
@@ -0,0 +1,346 @@
+        sum(max_over_time(ovn_db_cluster_outbound_connections_total{db_name="OVN_Northbound"}[5m]))
+        != count(max_over_time(kube_node_role{role="master"}[5m])) * (count(max_over_time(kube_node_role{role="master"}[5m]))-1)
+      for: 5m
+      labels:
+        severity: warning
+    - alert: OVNKubernetesSouthboundDatabaseOutboundConnectionMissing
+      annotations:
+        summary: OVN southbound database server(s) do not have expected number of outbound RAFT connections.
+        description: OVN southbound database server(s) do not have expected number of outbound connections for a RAFT cluster
+          which may indicate degraded OVN database high availability.
+      expr: |
+        # Expected sum of outbound connections is number of control plane nodes * number of control plane nodes minus one
+        # Without max_over_time, failed scrapes could create false negatives, see
+        # https://www.robustperception.io/alerting-on-gauges-in-prometheus-2-0 for details.
+        sum(max_over_time(ovn_db_cluster_outbound_connections_total{db_name="OVN_Southbound"}[5m]))
+        != count(max_over_time(kube_node_role{role="master"}[5m])) * (count(max_over_time(kube_node_role{role="master"}[5m]))-1)
+      for: 5m
+      labels:
+        severity: warning
+    # OVN northbound and southbound database performance alerts
+    - alert: OVNKubernetesNorthboundDatabaseCPUUsageHigh
+      annotations:
+        summary: OVN northbound database {{"{{"}} $labels.instance {{"}}"}} is greater than {{"{{"}} $value {{"}}"}} percent CPU usage for a period of time.
+        description: High OVN northbound CPU usage indicates high load on the networking control plane.
+      expr: (sum(rate(container_cpu_usage_seconds_total{container="nbdb"}[5m])) BY (instance, name) * 100) > 80
+      for: 15m
+      labels:
+        severity: info
+    - alert: OVNKubernetesSouthboundDatabaseCPUUsageHigh
+      annotations:
+        summary: OVN southbound database {{"{{"}} $labels.instance {{"}}"}} is greater than {{"{{"}} $value {{"}}"}} percent CPU usage for a period of time.
+        description: High OVN southbound CPU usage indicates high load on the networking control plane.
+      expr: (sum(rate(container_cpu_usage_seconds_total{container="sbdb"}[5m])) BY (instance, name) * 100) > 80
+      for: 15m
+      labels:
+        severity: info
+    # OVN northd functional alerts
+    - alert: OVNKubernetesNorthdInactive
+      annotations:
+        summary: No OVN northd is active. Networking is degraded.
+        description: No OVN northd is available within the high availability set. Networking control plane is degraded.
+      expr: |
+        # Without max_over_time, failed scrapes could create false negatives, see
+        # https://www.robustperception.io/alerting-on-gauges-in-prometheus-2-0 for details.
+        count(max_over_time(ovn_northd_status[5m]) == 1) != 1
```

Comment:
What does the metric `ovn_northd_status` mean here? In my cluster, I can see 3 of them and 1 has value 1, the others are 0.

Reply:
Specifies whether this instance of ovn-northd is standby(0) or active(1) or paused(2). Active holds the lock to write to SB DB. Standbys continue to get updates from NB DB but do not write to SB DB.

Reply:
Also, northd is in HA mode in OCP so thats why youre seeing 3 time series.